kuma/wiki/tests/test_admin.py

```diff
@@ -2,21 +2,170 @@
 from pyquery import PyQuery as pq
 
 from constance.test import override_config
+from django.contrib.admin import AdminSite
+from django.test import RequestFactory
 import requests_mock
 from waffle.models import Flag
 
 from kuma.core.urlresolvers import reverse
 from kuma.core.utils import urlparams
 from kuma.spam.akismet import Akismet
-from kuma.spam.constants import SPAM_SUBMISSIONS_FLAG, SPAM_URL, VERIFY_URL
+from kuma.spam.constants import HAM_URL, SPAM_SUBMISSIONS_FLAG, SPAM_URL, VERIFY_URL
 from kuma.users.tests import UserTestCase
 from kuma.users.models import User
-from kuma.wiki.models import RevisionAkismetSubmission
+from kuma.wiki.admin import DocumentSpamAttemptAdmin
+from kuma.wiki.models import DocumentSpamAttempt, RevisionAkismetSubmission
+
+
+@pytest.mark.spam
+class DocumentSpamAttemptAdminTestCase(UserTestCase):
+    fixtures = UserTestCase.fixtures + ['wiki/documents.json']
+    sample_data = """{
+        "blog_lang": "en_us",
+        "comment_type": "wiki-revision",
+        "referrer": "https://developer.mozilla.org/en-US/docs/new?slug=Web/CSS",
+        "user_ip": "192.168.10.1",
+        "blog_charset": "UTF-8",
+        "user_agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3)",
+        "comment_content": "Web_CSS\\nWeb_CSS\\n\\n<p>spam</p>\\nComment\\n\\n",
+        "comment_author": "viagra-test-123",
+        "comment_author_email": "viagra-test-123@example.com"
+    }"""
+
+    class FakeMessageStorage(object):
+        def __init__(self):
+            self.storage = []
+
+        def add(self, level, message, extra_tags):
+            self.storage.append((level, message, extra_tags))
```

Comment:
there is a `get_messages` test method in `kuma.core.tests` to do something similar